library/src/com/squareup/timessquare/CalendarPickerView.java

```diff
@@ -644,7 +663,16 @@ public void highlightDates(Collection<Date> dates) {
     }
 
     adapter.notifyDataSetChanged();
-    setAdapter(adapter);
```

Comment:
I removed it on purpose as I don't think it does anything at that point. Otherwise it is probably better to call `validateAndUpdate()`?